src/main/java/org/openjdk/shenandoah/ShenandoahVisualizer.java

```diff
@@ -22,79 +23,354 @@
  */
 package org.openjdk.shenandoah;
 
+import org.HdrHistogram.Histogram;
+
 import javax.swing.*;
+import javax.swing.event.ChangeEvent;
+import javax.swing.event.ChangeListener;
 import java.awt.*;
-import java.awt.event.ComponentAdapter;
-import java.awt.event.ComponentEvent;
-import java.awt.event.WindowAdapter;
-import java.awt.event.WindowEvent;
-import java.util.BitSet;
+import java.awt.event.*;
+import java.io.IOException;
+import java.util.ArrayList;
 import java.util.LinkedHashMap;
 import java.util.LinkedList;
+import java.util.List;
 import java.util.Map;
-import java.util.concurrent.Executors;
-import java.util.concurrent.ScheduledExecutorService;
-import java.util.concurrent.ScheduledFuture;
-import java.util.concurrent.TimeUnit;
+import java.util.concurrent.*;
 
 import static org.openjdk.shenandoah.RegionState.*;
 
 class ShenandoahVisualizer {
 
-    private static final int INITIAL_WIDTH = 1000;
-    private static final int INITIAL_HEIGHT = 800;
+    private static final int INITIAL_WIDTH = 2000;
+    private static final int INITIAL_HEIGHT = 1600;
+    private static final int KILO = 1024;
+    private static final String PLAYBACK = "Playback";
+    private static final String REALTIME = "Realtime";
+
+    static int value = 0;
+
+    private static ScheduledFuture<?> changeScheduleInterval(int n, ScheduledExecutorService service, ScheduledFuture<?> f, Runnable task) {
+        if (service == null || f == null) return null;
+        if (n > 0) {
+            boolean res = true;
+            if (f != null) {
+                res = f.cancel(true);
+            }
+            f = service.scheduleAtFixedRate(task, 0, n, TimeUnit.MILLISECONDS);
+            return f;
+        }
+        return null;
+    }
 
     public static void main(String[] args) throws Exception {
-        if (args.length < 1) {
-            System.err.println("missing VM identifier");
-            System.exit(-1);
+        // Command line argument parsing
```

Comment:
This method is way too big for my taste, and has way too many inline/anonymous classes. This could benefit from some refactoring and better structuring, IMO.

Reply:
I'll see what I can do here.